src/app/workflow/workflow.js

```diff
@@ -42,6 +42,9 @@ angular.module('app.workflow', [
           resolve: {
             workflow: function($stateParams, workflowManager) {
               return workflowManager.read($stateParams.workflowId);
+            },
+            workorders: function(workorderManager) {
```

Comment:
Rather than preload the list of workordrs on every page request (via this ui-router _resolve_ property), we could load them asynchronously as part of the _delete_ handler function.  Since the handler function is already async, we would just chain the promises, and not bother loading the workorder list for the majority of non-deleting page views.